src/lightning/pytorch/trainer/connectors/data_connector.py

```diff
@@ -400,14 +400,14 @@ def _check_dataloader_iterable(
         prefix = "train_" if trainer_fn == TrainerFn.FITTING else ""
         if not source.is_module():
             raise TypeError(
-                f"An invalid dataloader was passed to `Trainer.{trainer_fn}({prefix}dataloaders=...)`."
+                f"An invalid dataloader was passed to `Trainer.{trainer_fn.value}({prefix}dataloaders=...)`."
```

Comment:
I made the enum changes because in Python 3.11, the string representation of the enum changed.